app/src/main/java/e/josephmolina/saywhat/Speech/PollyClient.java

```diff
@@ -0,0 +1,62 @@
+package e.josephmolina.saywhat.Speech;
+
+import android.media.AudioManager;
+import android.media.MediaPlayer;
+import android.util.Log;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.services.polly.AmazonPollyPresigningClient;
+import com.amazonaws.services.polly.model.OutputFormat;
+import com.amazonaws.services.polly.model.SynthesizeSpeechPresignRequest;
+
+import java.io.IOException;
+import java.net.URL;
+
+import e.josephmolina.saywhat.Utils.Utils;
+import e.josephmolina.saywhat.language.Interpret;
+
+public class PollyClient {
+    private AmazonPollyPresigningClient pollyClient;
+    private Interpret interpret;
+    private final String ENGLISH_VOICE_ID = "Joey";
+    private final String SPANISH_VOICE_ID = "Mia";
+
+    public PollyClient() {
+        AWSCredentialsProvider credentialsProvider = AWSMobileClient.getInstance().getCredentialsProvider();
+        pollyClient = new AmazonPollyPresigningClient(credentialsProvider);
+        interpret = new Interpret();
+    }
+
+    private String getCorrespondingVoice(String text) {
+        String dominantLanguage = interpret.detectLanguage(text);
```

Comment:
Can we pass the language code in from the MainController, rather than making another request to detect the language?

Reply:
I looked into it but it would ruin the abstraction of just calling pollyclient.speak(text)
Do you have a suggestion on how to do this or list this as a high priority?